### PR TITLE
Architecture_Engine: Add CleanRoom modify method

### LINKED ISSUE #1611: Architecture_Engine: Create CleanRoom method to clean room perimeter
#### Description:

Similar to the CleanPanel method in Environment_Engine.
Functions within the method:
remove vertices
simplify curves
recompute edges

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #1611 

### Test files
@BingWangUS or @michaelhoehn to provide

### CODE REVIEW

**Comment by BingWangUS** [comment]
Test script here.
[Debug_CleanRoom.zip](https://github.com/BHoM/BHoM_Engine/files/4462761/Debug_CleanRoom.zip)

Function wise, it perform as expected.

File: Architecture_Engine/Modify/CleanRoom.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a room that has been cleaned of short segments and insignificant vertices in its perimeter curve")]
+        [Input("room", "An architecture room that will be cleaned")]
+        [Input("angleTolerance", "The tolerance of the angle that defines a straight line. Default is set to the value defined by BH.oM.Geometry.Tolerance.Angle")]
+        [Input("minimumSegmentLength", "The length of the smallest allowed segment. Segments smaller than this will be removed. Default is set to the value defined by BH.oM.Geometry.Tolerance.Distance")]
+        [Output("cleanedRoom", "A room that has been cleaned")]
+        public static Room CleanPanel(this Room room, double angleTolerance = Tolerance.Angle, double minimumSegmentLength = Tolerance.Distance)
```
Comment (BingWangUS) [inline Architecture_Engine/Modify/CleanRoom.cs:43]: Should this be CleanRoom?

**Comment by FraserGreenroyd** [comment]
@BingWangUS good catch, fixed

**Comment by BingWangUS** [comment]
This should also close #1594

**Pull request merged**